feature/home/src/main/java/com/mashup/dorabangs/feature/home/PostDataManager.kt

```diff
@@ -0,0 +1,87 @@
+package com.mashup.dorabangs.feature.home
+
+import com.mashup.dorabangs.core.designsystem.component.chips.FeedUiModel.FeedCardUiModel
+import com.mashup.dorabangs.domain.model.Folder
+import com.mashup.dorabangs.domain.model.Post
+import com.mashup.dorabangs.domain.model.Posts
+import com.mashup.dorabangs.feature.model.PagingInfo
+import com.mashup.dorabangs.feature.model.toUiModel
+import java.util.concurrent.ConcurrentHashMap
+
+class PostDataManager(
+    private val getPost: suspend (String, PagingInfo) -> Posts,
+) {
+    private val pagingInfoCache = ConcurrentHashMap<String, PagingInfo>()
+    private val postIdCache = ConcurrentHashMap<String, List<String>>()
+    private val postDataCache = ConcurrentHashMap<String, FeedCardUiModel>()
+
+    suspend fun fetchPostData(key: String, folderList: List<Folder>): List<FeedCardUiModel> {
+        return if (postIdCache[key].isNullOrEmpty()) {
+            fetchRemotePostData(key, folderList)
+        } else {
+            postIdCache[key]
+                ?.mapNotNull { postId -> postDataCache[postId] }
+                .orEmpty()
+        }
+    }
+
+    suspend fun fetchRemotePostData(key: String, folderList: List<Folder>): List<FeedCardUiModel> {
+        val pagingInfo = getPagingInfo(key)
+        return getPost(key, pagingInfo)
+            .updateNextPagingInfo(key)
+            .items
+            .toUIModel(folderList)
+            .caching(key)
+    }
+
+    fun getPostIdList(key: String) = postIdCache[key].orEmpty()
+
+    fun getPost(postId: String) = postDataCache[postId]
+
+    fun getHasNext(key: String) = pagingInfoCache[key]?.hasNext ?: false
+
+    fun updatePostCache(post: FeedCardUiModel) {
+        updatePostCache(post.postId, post)
+    }
+
+    fun updatePostCache(postId: String, post: FeedCardUiModel) {
+        postDataCache[postId] = post
+    }
+
+    fun updatePostIdCache(key: String, idList: List<String>) {
+        postIdCache[key] = idList
+    }
+
+    fun setMustRefreshData(key: String) {
+        postIdCache.remove(key)
+        pagingInfoCache.remove(key)
+    }
+
+    private fun getPagingInfo(key: String): PagingInfo {
+        return pagingInfoCache[key] ?: PagingInfo.getDefault(key).apply {
+            pagingInfoCache[key] = this
+        }
+    }
+
+    private fun Posts.updateNextPagingInfo(key: String): Posts {
+        val pagingInfo = getPagingInfo(key)
```

Comment:
이거 안하고 그냥 pagingInfo를 넘겼어도 됐지 않나 싶어서
```suggestion
    private fun Posts.updateNextPagingInfo(key: String, pagingInfo: PagingInfo): Posts {
```

Reply:
그러면 함수명에 맞게 진짜 updateNextPagingInfo만 수행하지 않나 싶어서용

Reply:
저 키도 페이징 인포 가져오려고 하는거면 그 위에서 받아서 내려오죠 ~